<?xml version="1.0"?>
<launch>
    <!-- 无人机命名空间参数 -->
    <arg name="vehicle_ns" default="iris_0"/>
    
    <!-- 加载所有航点参数从 params.yaml -->
    <rosparam file="$(find mission_master_pkg)/config/params.yaml" command="load"/>
    
    <!-- 覆盖命名空间参数（如需要不同的无人机） -->
    <param name="vehicle_namespace" value="/$(arg vehicle_ns)"/>
    
    <!-- 启动 Mission Master 节点 -->
    <node pkg="mission_master_pkg" type="mission_master_node" name="mission_master" output="screen"/>
    
    <!-- 可选：启动 RViz 可视化 -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find mission_master_pkg)/config/rviz_config.rviz"/> -->
</launch>
